src/tests/incomplete_utf8.pl

```diff
@@ -0,0 +1,111 @@
+:- module(incomplete_utf8_tests, []).
+:- use_module(test_framework).
+:- use_module(library(charsio)).
+:- use_module(library(process)).
+:- use_module(library(format)).
+
+test("get_char/2 after get_n_chars/4 timeout mid-char", (
+    atom_chars('python3', Py),
+    atom_chars('-c', C),
+    atom_chars('import sys,time; sys.stdout.buffer.write(b\"\\xf0\"); sys.stdout.buffer.flush(); time.sleep(0.5); sys.stdout.buffer.write(b\"\\x9f\\x92\\x9cAB\"); sys.stdout.buffer.flush()', Cmd),
+    process_create(Py, [C, Cmd], [stdout(pipe(Out))]),
+    get_n_chars(Out, _, Chars1, 100),
+    Chars1 = [],
+    get_char(Out, C1),
+    C1 = '💜',
```

Comment:
A stronger test would be using `(==)/2` here, it ensures that the terms are really identical.

(In a wrong implementation of `get_char/2`, `C1` may be a variable after success, and the unification then succeeds.)

Reply:
The difference between these two predicates is killing me in my scheduler implementation :sweat_smile:

Reply:
https://github.com/mthom/scryer-prolog/pull/3136/commits/ede8e197be2ef7b51939534b9a812cefd08a4a0f